web-ui/router.ml

```diff
@@ -203,6 +203,14 @@ let github_routes github =
           ~hash:(Dream.param request "hash")
           ~variant:(Dream.param request "variant")
           github);
+    Dream.get "/api/github/:org/:repo/commit/:hash" (fun request ->
+        let org = Dream.param request "org" in
+        let repo = Dream.param request "repo" in
+        let hash = Dream.param request "hash" in
+        Controller.Api_github.list_steps ~org ~repo ~hash
+          ~step_route_prefix:
+            (Fmt.str "/github/%s/%s/commit/%s/variant" org repo hash)
+          github);
```

Comment:
Essentially because `list_steps` doesn't know which git_forge its working with, but more importantly, a controller method should not need to know anything about routing rules.

The router module is responsible for setting up routing rules. If our routes change, it would be good to have this change be localised to the router. 

It would be good to have helper functions available to the controllers and views so that routes/paths can be asked for as needed. I suggest we do this if/when there is another instance of having to pass the path around like I am doing now (I agree that it is ugly btw).

Reply:
Agreed, thanks for the precision :+1: